src/main/scala/com/mwz/sonar/scala/sensor/Measures.scala

```diff
@@ -33,7 +38,7 @@ object Measures {
   }
 
   def countMethods(tokens: List[Token]): Int = {
```

Comment:
I had been thinking about these count methods, if you check the code of `ScalaSensor`, you can see that the same _tokens_ are used to call all these four methods, so the same `List` gets traversed 4 times. Maybe we could make one method that make the four counts in one pass and return some `case class` with the output. However, I don't know if that would end with a method that is difficult to understand and difficult to test. So at the end, is this trade-off  between readability and simplicity VS performance, I have to admit that a scan of my project doesn't take much time, but my project have no more than 50 files, so I'm wondering what happens with big projects.

By the way, I come from a HPC background, so I tend to over-optimize things.
In any case, that should be addressed in another PR, I just wanted to kick off a discussion about it.

Reply:
I wouldn't prematurely optimise this unless there are any valid reasons. It would be interesting, though to see how long the analysis would take for a large project - the largest I've tried so far was about 7k lines of code and 150 files and the full analysis including scapegoat takes about 10 seconds.

Reply:
Oh ok, fast enough :+1: